### PR TITLE
Qm adhras

### PR DESCRIPTION
**Pull request by Bijaya-Maitri**
## Describe your changes
Please include a summary of any changes.

## How has this been tested?
Please describe the tests you ran to verify your changes.  Provide instructions or code to reproduce output.

## Reviewer focus
Please summarize the specific items you’d like the reviewer(s) to look into.

## Checklist before requesting a review
- [ ] I have added at least one Github label to this PR (bug, enhancement, breaking change,...)
- [ ] My code follows [style guidelines](https://thetuvaproject.com/guides/contributing/style-guide)
- [ ] (New models) [YAML files](https://github.com/tuva-health/tuva/blob/main/models/hcc_suspecting/hcc_suspecting_models.yml) are categorized by sub folder and models listed in alphabetical order
- [ ] (New models) I have added a [config](https://github.com/tuva-health/tuva/blob/main/models/hcc_suspecting/final/hcc_suspecting__list.sql) to each new model to enable it for claims and/or clinical data
- [ ] (New models) I have added the variable `tuva_last_run` to the final output
- [ ] (Optional) I have recorded a Loom to explain this PR

### Package release checklist
- [ ] I have updated [dbt docs](https://www.notion.so/tuvahealth/Building-dbt-Docs-16df2f00df244f29b9d6756d8adbc2d9)
- [ ] I have updated the version number in the `dbt_project.yml`

### CODE REVIEW

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+)
+
+, valid_hospice_palliative as (
+
+  select
+      patient_id
+    , exclusion_date
+    , exclusion_reason
+  from {{ref('quality_measures__int_shared_exclusions_hospice_palliative')}}
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql]:
```suggestion
  from {{ ref('quality_measures__int_shared_exclusions_hospice_palliative') }}
```

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+)
+
+, sacubitril_codes as (
+
+    select
+          code
+        , code_system
+        , concept_name
+    from {{ ref('quality_measures__value_sets') }}
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql]: ref value set only in a cte and then reference this cte for filtering both concepts

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+, measure_exclusions as (
+
+    select 
+          exclusions.patient_id
+        , exclusion_date
+        , exclusion_reason
+    from exclusions
+    inner join denominator
+      on exclusions.patient_id = denominator.patient_id
+)
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql]:
```suggestion
      on exclusions.patient_id = denominator.patient_id
      
)
```

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+, add_data_types as (
+
+  select
+      distinct
+          cast(patient_id as {{ dbt.type_string() }}) as patient_id
+        , cast(exclusion_date as date) as exclusion_date
+        , cast(exclusion_reason as {{ dbt.type_string() }}) as exclusion_reason
+        , 1 as exclusion_flag
+  from measure_exclusions
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql]:
```suggestion
    select
        distinct
            cast(patient_id as {{ dbt.type_string() }}) as patient_id
          , cast(exclusion_date as date) as exclusion_date
          , cast(exclusion_reason as {{ dbt.type_string() }}) as exclusion_reason
          , 1 as exclusion_flag
    from measure_exclusions
```

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_long.sql [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+)
+
+, numerator as (
+
+    select
+          patient_id
+        , evidence_date
+        , evidence_value
+    from {{ ref('quality_measures__int_adhras_numerator') }}
+    where numerator_flag = 1
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_long.sql]: this is not be needed as all records in numerator have this flag 1.
```suggestion
```

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_numerator.sql [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+*/
+
+, ranked_patient as (
+
+    select
+          patient_id
+        , dispensing_date
+        , ndc_code
+        , days_supply
+        , dense_rank() over (partition by patient_id order by dispensing_date) as row_number
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_numerator.sql]: Update row_number alias to dense_rank wherever referenced as well
```suggestion
        , dense_rank() over (partition by patient_id order by dispensing_date) as dense_rank
```

File: models/quality_measures/quality_measures_models.yml [hunk 7]
```diff
@@ -2444,6 +2448,159 @@ models:
...
+        description: >
+          Date of event or service that places patient in the numerator.
+      - name: evidence_value
+        description: >
+          Observed evidence value for the patients in the numerator.
+      - name: numerator_flag
+        description: >
+          The numerator reflects the subset of patients in the denominator 
+          for whom a particular service has been provided or for whom a 
+          particular outcome has been achieved.
```
Comment (utsavpaudel) [inline models/quality_measures/quality_measures_models.yml:2604]: add documentation of adherence field in numerator

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+          medication.patient_id
+        , medication.dispensing_date as exclusion_date
+        , sacubitril_codes.concept_name as exclusion_reason
+    from {{ ref('quality_measures__stg_core__medication') }} as medication
+    inner join sacubitril_codes
+      on medication.source_code = sacubitril_codes.code 
+        and medication.source_code_type = sacubitril_codes.code_system
+    where medication.dispensing_date between {{ performance_period_begin }} and {{ performance_period_end }}
+
+)
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql]: dont use this clinical table, data source is pharmacy claims in measure specification document

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql (hunk 2 above)
Comment (Bijaya-Maitri) [reply models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql]: Done

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql (hunk 1 above)
Comment (Bijaya-Maitri) [reply models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_exclusions.sql]: Done

**Pull request merged**